src/halfpipe/tui/working_directory/base.py

```diff
@@ -0,0 +1,525 @@
+
+        # The philosophie here is that we copy the data from the existing spec file to the context cache and then create
+        # corresponding widgets. Through these widgets then it should be possible to further modify the spec file.
+        # The created widgets should avoid using step and meta classes upon creation as these triggers various user choice
+        # modals.
+        if self.existing_spec is not None:
+            self.app.get_widget_by_id("input_data_content").toggle_bids_non_bids_format(False)
+            spreadsheet_counter = 0
+            self.event_file_objects = []
+            self.atlas_file_objects = []
+            self.seed_map_file_objects = []
+            self.spatial_map_file_objects = []
+
+            preprocessing_widget.default_settings = self.existing_spec.global_settings
+            preprocessing_widget = preprocessing_widget.refresh(recompose=True)
+
+            for f in self.existing_spec.files:
+                # In the spec file, subject is abbreviated to 'sub' and atlas, seed and map is replaced by desc,
+                # here we replace it back for the consistency.
+                f.__dict__["path"] = f.__dict__["path"].replace("{sub}", "{subject}")
+                if f.datatype == "bids":
+                    ctx.cache["bids"]["files"] = f.path
+                    data_input_widget.get_widget_by_id("data_input_file_browser").update_input(f.path)
+                    # this is the function used when we are loading bids data files, in also checks if the data
+                    # folder contains bids files, and if yes, then it also extracts the tasks (images)
+                    path_test_for_bids(f.path)
+                    self.app.get_widget_by_id("input_data_content").toggle_bids_non_bids_format(True)
+                # need to create a FileItem widgets for all non-bids files
+                elif f.datatype == "spreadsheet":
+                    ctx.cache["__spreadsheet_file_" + str(spreadsheet_counter)]["files"] = f
+                    spreadsheet_counter += 1
+                elif f.suffix == "bold":
+                    message_dict = {i: [str(f.metadata[i])] for i in f.metadata if i == "repetition_time"}
+                    widget_name = await data_input_widget.add_bold_image(
+                        load_object=f, message_dict=message_dict, execute_pattern_class_on_mount=False
+                    )
+                    ctx.cache[widget_name]["files"] = f
+                elif f.suffix == "T1w":
+                    widget_name = await data_input_widget.add_t1_image(
+                        load_object=f, message_dict=None, execute_pattern_class_on_mount=False
+                    )
+                    ctx.cache[widget_name]["files"] = f
+                elif f.suffix == "events":
+                    self.event_file_objects.append(f)
+                elif f.suffix == "atlas":
+                    f.__dict__["path"] = f.__dict__["path"].replace("{desc}", "{atlas}")
```

Comment:
Why are you using the `__dict__` syntax? Does directly setting the `path` attribute not work somehow?

Reply:
Couldn't you do `f.path = f.path.replace("{desc}", "{atlas}"`?

Reply:
yes, done